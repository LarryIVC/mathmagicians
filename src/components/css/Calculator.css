.main-cont {
  display: flex;
  flex-direction: row;
  gap: 10px;
  justify-content: space-between;
  width: 80%;
  margin: 0 auto;
}

.calc-container {
  background-color: #edebeb;
  width: 400px;
  display: flex;
  flex-direction: column;
  gap: 1px;
  height: auto;
  border: 1px solid #858693;
  font-weight: bold;
}

.row {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 60px;
  gap: 1px;
}

.last-row {
  display: grid;
  grid-template-columns: 2fr 1fr 1fr;
  grid-column: 1 / span 2;
  gap: 1px;
}
