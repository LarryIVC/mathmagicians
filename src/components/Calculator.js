import React, { useState } from 'react';
import Display from './Display';
import Key from './Key';
import './css/Calculator.css';
import calculate from './logic/calculate';

const initialValues = {
  total: 0,
  next: null,
  operation: null,
};

const Calculator = () => {
  const [state, setState] = useState(initialValues);

  const handleClick = (buttonName) => {
    setState((prevState) => {
      const newResult = calculate(prevState, buttonName.target.textContent);
      return { ...prevState, ...newResult };
    });
  };

  return (
    <div className="main-cont">
      <article>
        <h2>Let&apos;s do some math!</h2>
      </article>
      <article>
        <div className="calc-container">
          <div className="row">
            <Display input={state} />
          </div>
          <div className="row">
            <Key manageClick={handleClick}>AC</Key>
            <Key manageClick={handleClick}>+/-</Key>
            <Key manageClick={handleClick}>%</Key>
            <Key manageClick={handleClick}>รท</Key>
          </div>
          <div className="row">
            <Key manageClick={handleClick}>7</Key>
            <Key manageClick={handleClick}>8</Key>
            <Key manageClick={handleClick}>9</Key>
            <Key manageClick={handleClick}>x</Key>
          </div>
          <div className="row">
            <Key manageClick={handleClick}>4</Key>
            <Key manageClick={handleClick}>5</Key>
            <Key manageClick={handleClick}>6</Key>
            <Key manageClick={handleClick}>-</Key>
          </div>
          <div className="row">
            <Key manageClick={handleClick}>1</Key>
            <Key manageClick={handleClick}>2</Key>
            <Key manageClick={handleClick}>3</Key>
            <Key manageClick={handleClick}>+</Key>
          </div>
          <div className="row last-row">
            <Key manageClick={handleClick}>0</Key>
            <Key manageClick={handleClick}>.</Key>
            <Key manageClick={handleClick}>=</Key>
          </div>
        </div>
      </article>
    </div>
  );
};

export default Calculator;
