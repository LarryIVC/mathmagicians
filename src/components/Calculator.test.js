import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calculator from './Calculator';

test('renders Calculator component', () => {
  const { container } = render(<Calculator />);
  expect(container).toMatchSnapshot();
});

test('handles button click and updates state', () => {
  const { getByText } = render(<Calculator />);

  fireEvent.click(getByText('1'));
  fireEvent.click(getByText('2'));
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('3'));
  fireEvent.click(getByText('='));

  expect(getByText('15')).toBeInTheDocument();
});
