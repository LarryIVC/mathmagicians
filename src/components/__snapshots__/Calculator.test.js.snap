// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator component 1`] = `
<div>
  <div
    class="main-cont"
  >
    <article>
      <h2>
        Let's do some math!
      </h2>
    </article>
    <article>
      <div
        class="calc-container"
      >
        <div
          class="row"
        >
          <div
            class="Display"
          >
            0
          </div>
        </div>
        <div
          class="row"
        >
          <button
            class="key "
            type="button"
          >
            AC
          </button>
          <button
            class="key "
            type="button"
          >
            +/-
          </button>
          <button
            class="key "
            type="button"
          >
            %
          </button>
          <button
            class="key orange"
            type="button"
          >
            รท
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="key "
            type="button"
          >
            7
          </button>
          <button
            class="key "
            type="button"
          >
            8
          </button>
          <button
            class="key "
            type="button"
          >
            9
          </button>
          <button
            class="key orange"
            type="button"
          >
            x
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="key "
            type="button"
          >
            4
          </button>
          <button
            class="key "
            type="button"
          >
            5
          </button>
          <button
            class="key "
            type="button"
          >
            6
          </button>
          <button
            class="key orange"
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="key "
            type="button"
          >
            1
          </button>
          <button
            class="key "
            type="button"
          >
            2
          </button>
          <button
            class="key "
            type="button"
          >
            3
          </button>
          <button
            class="key orange"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="row last-row"
        >
          <button
            class="key "
            type="button"
          >
            0
          </button>
          <button
            class="key "
            type="button"
          >
            .
          </button>
          <button
            class="key orange"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </article>
  </div>
</div>
`;
